SPIFlash.cpp

```diff
@@ -270,6 +271,9 @@ void SPIFlash::blockErase4K(uint32_t addr) {
 
 /// erase a 32Kbyte block
 void SPIFlash::blockErase32K(uint32_t addr) {
+   if (readDeviceId() == 0x0102)  // check if Spansion S25FL032P 
```

Comment:
Unfortunately we like to keep separation of concerns in funcitons like this one and not write specific code for a device. So I would suggest moving this in an override function in an inherited class.